response_operations_ui/templates/collection_exercise/ce-sample-ci-section.html

```diff
@@ -1,80 +1,97 @@
 <section class="ce-section ce-details">
     {% set surveyEditPermission = hasPermission('surveys.edit') %}
+    {% set tblHeaders =  
```

Comment:
We have a bit of repetition of if statements, we should try to reduce them to the minimum, something like 
```  
   {% set tblHeaders =  
        [ 
            { 
                "value": "Sample & Collection instruments (CI)",
                "thClasses": "ons-u-fs-r--b"
            }
        ]
    %}
  
    {% set surveyTableData = {
      "variants": ['compact', 'responsive'],
      "id": "sample-table",
      "caption": 'Sample summary',
      "hideCaption": true,
      "ths": tblHeaders,
    } %}
  
    {% if sample %}
      {% if sample.ingestDateTime %}
        {% do tblHeaders.append(
            {
                "value": "",
                "thClasses": "ons-u-ta"
            }
         )%}
        {% do tblHeaders.append(
            {
                "value": "Uploaded " + '<em>' + sample.ingestDateTime[:-8] + '</em>' + sample.ingestDateTime[-8:],
                "thClasses": "ons-u-fs-r",
                "numeric": "true"
            }
         )%}
        {% set surveyTableDataRows = [
                {
                    "tds": [
                        {
                            "value": "Total businesses",
                            "tdClasses": "ons-u-ta"
                        },
                        {
                            "value": sample.totalSampleUnits,
                            "name": "total-instruments",
                            "tdClasses": "ons-u-ta-right"
                        },
                        {
                            "value": ' ',
                            "name": "upload-sample-time",
                        }
                    ]
                },
                {
                    "tds": [
                        {
                            "value": "Collection instruments",
                            "tdClasses": "ons-u-ta"
                        },
                        {
                            "value": sample.expectedCollectionInstruments,
                            "name": "total-instruments",
                            "tdClasses": "ons-u-ta-right"
                        },
                        {
                            "value": ' ',
                            "name": "upload-sample-time",
                        }
                    ]
                }
            ] %}
            
        {% do surveyTableData | setAttribute("trs", surveyTableDataRows) %}
      {% endif %}
  
      {{ onsTable(surveyTableData) }}
  
      {% if 'INIT' == sample.state %}
            <div class="ons-u-mb-m">
                <div class="ons-u-fs-r">
                    Loading&hellip; <span class="ons-u-ml-xs"><a href="/surveys/{{survey.shortName}}/{{ce.exerciseRef}}/view-sample-ci?show_msg=true">Refresh to see progress</a></span>
                </div>
            </div>
      {% elif 'FAILED' == sample.state %}
           <div class="ons-u-mb-xl">
                {% call
                    onsPanel({
                        "type": "error",
                        "classes": "ons-u-mb-l"
                    })
                %}
                    {{ sample.notes }}
                {% endcall %}
            </div>
      {% endif %}
    {% else %}
      {{ onsTable(surveyTableData) }}
    {% endif %}

Reply:
I did take a bit of liberty with ingestDateTime, which can be changed back to live/init